### PR TITLE
Backup prod dynamodb

### PR DESCRIPTION
**Pull request by rich-nguyen**
This schedules an export of prod dynamo db tables to dev.

### CODE REVIEW

File: cloudformation/campaign-central-backups.yaml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+          Attributes:
+            -
+              Key: "description"
+              StringValue: "DynamoDB TableName2"
+            -
+              Key: "type"
+              StringValue: "String"
+      ParameterValues:
+        -
+          Id: "myTableName1"
```
Comment (LATaylor-guardian) [inline cloudformation/campaign-central-backups.yaml]: As above.

File: cloudformation/campaign-central-backups.yaml [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+          StringValue:
+            Ref: "SourceTableName"
+        -
+          Id: "myTableName2"
```
Comment (LATaylor-guardian) [inline cloudformation/campaign-central-backups.yaml]: As above.

File: cloudformation/campaign-central-backups.yaml [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+              Key: "description"
+              StringValue: "DynamoDB read throughput ratio"
+            -
+              Key: "type"
+              StringValue: "Double"
+            -
+              Key: "default"
+              StringValue: "0.2"
+        -
+          Id: "myTableName1"
```
Comment (LATaylor-guardian) [inline cloudformation/campaign-central-backups.yaml]: Can this have a more descriptive name?

File: cloudformation/campaign-central-backups.yaml [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+          Attributes:
+            -
+              Key: "description"
+              StringValue: "DynamoDB TableName1"
+            -
+              Key: "type"
+              StringValue: "String"
+        -
+          Id: "myTableName2"
```
Comment (LATaylor-guardian) [inline cloudformation/campaign-central-backups.yaml]: As above.

**Comment by LATaylor-guardian** [comment]
👍

File: cloudformation/campaign-central-backups.yaml (hunk 3 above)
Comment (rich-nguyen) [reply cloudformation/campaign-central-backups.yaml]: oh yeah, that's when i thought i was going to put all tables in one cfn template. will update

File: cloudformation/campaign-central-backups.yaml [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  DestinationTableName:
+    Description: DynamoDB tableName to write to.
+    Type: String
+    Default: campaign-central-DEV-campaigns
+Resources:
+  CampaignCentralDynamoDBBackup:
+    Type: "AWS::DataPipeline::Pipeline"
+    Properties:
+      Name: CampaignCentralDynamoDBBackup
+      Description: "Pipeline to backup DynamoDB data to S3"
```
Comment (kelvin-chappell) [inline cloudformation/campaign-central-backups.yaml:17]: `... to backup PROD DynamoDB data to DEV` ?

**Review by kelvin-chappell — COMMENTED** [review]
Never seen this data pipeline before.  It's much better than the way I was going to do it.

File: cloudformation/campaign-central-backups.yaml (hunk 5 above)
Comment (rich-nguyen) [reply cloudformation/campaign-central-backups.yaml:17]: The template is parameterised, so it could be any source table to any destination table!

**Comment by rich-nguyen** [comment]
@kelvin-chappell Yes, although I couldn't find a tidy way to clean up the table, so duplicates will arrive every week. It's the next step. After that, we can easily scale the template to update the other campaign central tables.

**Pull request merged**